projects/packages/scheduled-updates/src/class-scheduled-updates.php

```diff
@@ -91,6 +91,8 @@ public static function init() {
 		// Active flag saving.
 		add_action( 'add_option_' . Scheduled_Updates_Active::OPTION_NAME, $callback );
 		add_action( 'update_option_' . Scheduled_Updates_Active::OPTION_NAME, $callback );
+
+		add_filter( 'pre_schedule_event', array( __CLASS__, 'clear_cron_cache_pre' ) );
```

Comment:
Where does the filter get applied?

Reply:
Inside the `wp_schedule_event` function.

Reply:
Gotcha, thanks. In the callback, should we check that the event hook is our hook before clearing caches? Or does it need to run for every event scheduling?

Reply:
No because the problem is with the other events that obtain the wrong cacye, not ours. It will run for the first event that will be queued the first 10 seconds after our.